packages/uikit-workshop/src/scripts/components/modal-styleguide.js

```diff
@@ -22,12 +22,14 @@ export const modalStyleguide = {
     // go through the panel toggles and add click event to the pattern extra toggle button
     const toggles = document.querySelectorAll('.pl-js-pattern-extra-toggle');
 
-    toggles.forEach(toggle => {
-      toggle.addEventListener('click', e => {
-        const patternPartial = toggle.getAttribute('data-patternpartial');
-        modalStyleguide.toggle(patternPartial);
+    if (toggles.length > 0) {
```

Comment:
@shaal could you replace the `forEach` completely with a `for` loop by index? According the docs, `querySelectorAll` returns a `NodeList` object and can be accessed by index.

https://www.w3schools.com/js/js_htmldom_nodelist.asp

See the example on iterating in the web page above.

Reply:
@JosefBredereck even if you overhauled this PR, I think my remark still stands that it would be safer to loop over the `NodeList` object by index without needing the additional surronding `if` statement.

Reply:
As you see in the code I also changed that.

Reply:
I approved the PR. But given the condition of the `for` loop, I think the surrounding `if` statement is not needed anymore and can be removed.